src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,11 +31,20 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        Pattern pattern = Pattern.compile(ADDRESS_VALIDATION_REGEX);
+        Matcher matcher = pattern.matcher(trimmedAddress);
+
+        if(matcher.find()){
+            this.block = new Block(matcher.group(1));
+            this.street = new Street(matcher.group(3));
+            this.unit = new Unit(matcher.group(5));
+            this.postalCode = new PostalCode(matcher.group(7));
+            this.isPrivate = isPrivate;
+        }
+        else {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        this.value = this.toString();
```

Comment:
No need to save the input address in `this.value` as it may cause inconsistency due to duplication.